flask_admin/contrib/mongoengine/view.py

```diff
@@ -343,7 +343,7 @@ def scaffold_sortable_columns(self):
 
         for n, f in self._get_model_fields():
             if type(f) in SORTABLE_FIELDS:
-                if self.column_display_pk or type(f) != mongoengine.ObjectIdField:
+                if self.column_display_pk or not isinstance(f, mongoengine.ObjectIdField):
```

Comment:
These are good improvements but they belong in a separate PR. Since they're not required by your feature, they just make it harder to review.